src/frontend/data/queries/index.spec.tsx

```diff
@@ -0,0 +1,628 @@
+import fetchMock from 'fetch-mock';
+import React from 'react';
+import { QueryClient, QueryClientProvider, setLogger } from 'react-query';
+import { renderHook } from '@testing-library/react-hooks';
+
+import {
+  useCreateVideo,
+  useOrganization,
+  usePlaylist,
+  usePlaylists,
+  useThumbnail,
+  useTimedTextTracks,
+  useVideo,
+  useVideos,
+} from './index';
+import {
+  organizationMockFactory,
+  playlistMockFactory,
+  thumbnailMockFactory,
+  timedTextMockFactory,
+  videoMockFactory,
+} from '../../utils/tests/factories';
+
+setLogger({
+  // tslint:disable-next-line:no-console
+  log: console.log,
+  warn: console.warn,
+  // no more errors on the console
+  error: () => {},
+});
+
+jest.mock('../appData', () => ({
+  appData: {
+    jwt: 'some token',
+  },
+}));
+
+jest.mock('../../utils/errors/report', () => ({
+  report: jest.fn(),
+}));
+
+describe('queries', () => {
+  afterEach(() => {
+    fetchMock.restore();
+    jest.resetAllMocks();
+  });
+
+  describe('useOrganization', () => {
+    it('requests the resource', async () => {
+      const organization = organizationMockFactory();
+      fetchMock.mock(`/api/organizations/${organization.id}/`, organization);
+      const queryClient = new QueryClient();
+
+      const wrapper = ({ children }: Element) => (
+        <QueryClientProvider client={queryClient}>
+          {children}
+        </QueryClientProvider>
+      );
+
+      const { result, waitFor } = renderHook(
+        () => useOrganization(organization.id),
+        { wrapper },
+      );
+      await waitFor(() => result.current.isSuccess);
+
+      expect(fetchMock.calls().length).toEqual(1);
+      expect(fetchMock.lastCall()![0]).toEqual(
+        `/api/organizations/${organization.id}/`,
+      );
+      expect(fetchMock.lastCall()![1]).toEqual({
+        headers: {
+          Authorization: 'Bearer some token',
+          'Content-Type': 'application/json',
+        },
+      });
+      expect(result.current.data).toEqual(organization);
+      expect(result.current.status).toEqual('success');
+    });
+
+    it('fails to get the resource', async () => {
+      const organization = organizationMockFactory();
+      fetchMock.mock(`/api/organizations/${organization.id}/`, 404);
+      const queryClient = new QueryClient({
+        defaultOptions: {
+          queries: {
+            retry: false,
+          },
+        },
+      });
+
+      const wrapper = ({ children }: Element) => (
+        <QueryClientProvider client={queryClient}>
+          {children}
+        </QueryClientProvider>
+      );
+
+      const { result, waitFor } = renderHook(
+        () => useOrganization(organization.id),
+        { wrapper },
+      );
+
+      await expect(
+        waitFor(() => result.current.isSuccess),
+      ).rejects.toThrowError('Timed out in waitFor after 1000ms.');
```

Comment:
![](https://media.giphy.com/media/DvyLQztQwmyAM/giphy.gif)

Reply:
you should use `waitForNextUpdate` instead of waiting the `waitFor` failure. https://react-hooks-testing-library.com/usage/advanced-hooks#async

Reply:
or `await waitFor(() => result.current.isError);`